common/lib/xmodule/xmodule/tests/test_services.py

```diff
@@ -163,3 +170,98 @@ class TestTeamsConfigurationService(ConfigurationServiceBaseClass):
+        super().setUpClass()
+
+        course_key = CourseLocator('test', 'sandbox_test', '2021_01')
+        cls.sandbox_service = SandboxService(course_id=course_key, contentstore=contentstore)
+        cls.zipfile = upload_file_to_course(
+            course_key=course_key,
+            contentstore=cls.sandbox_service.contentstore(),
+            source_file=cls.PYTHON_LIB_SOURCE_FILE,
+            target_filename=cls.PYTHON_LIB_FILENAME,
+        )
+
+    @staticmethod
+    def validate_can_execute_unsafe_code(context_key, expected_result):
+        sandbox_service = SandboxService(course_id=context_key, contentstore=None)
+        assert expected_result == sandbox_service.can_execute_unsafe_code()
+
+    @ddt.data(
+        CourseLocator('edX', 'notful', 'empty'),
+        LibraryLocator('edY', 'test_bank'),
+    )
+    @override_settings(COURSES_WITH_UNSAFE_CODE=['edX/full/.*', 'library:v1-edX+.*'])
+    def test_sandbox_exclusion(self, context_key):
+        """
+        Test to make sure that a non-match returns false
+        """
+        self.validate_can_execute_unsafe_code(context_key, False)
+
+    @ddt.data(
+        CourseKey.from_string('edX/full/2012_Fall'),
+        CourseKey.from_string('edX/full/2013_Spring'),
+    )
+    @override_settings(COURSES_WITH_UNSAFE_CODE=['edX/full/.*'])
+    def test_sandbox_inclusion(self, context_key):
+        """
+        Test to make sure that a match works across course runs
+        """
+        self.validate_can_execute_unsafe_code(context_key, True)
+        self.validate_can_execute_unsafe_code(LibraryLocator('edX', 'test_bank'), False)
+
+    @ddt.data(
+        CourseLocator('edX', 'full', '2012_Fall'),
+        CourseLocator('edX', 'full', '2013_Spring'),
+        LibraryLocator('edX', 'test_bank'),
+    )
+    def test_courselikes_with_unsafe_code_default(self, context_key):
+        """
+        Test that the default setting for COURSES_WITH_UNSAFE_CODE is an empty setting,
+        i.e., we don't use @override_settings in these tests
+        """
+        self.validate_can_execute_unsafe_code(context_key, False)
+
+    @override_settings(PYTHON_LIB_FILENAME=PYTHON_LIB_FILENAME)
+    def test_get_python_lib_zip(self):
+        assert self.sandbox_service.get_python_lib_zip() == self.zipfile
+
+    def test_no_python_lib_zip(self):
+        assert self.sandbox_service.get_python_lib_zip() is None
+
+
+class CacheServiceTest(TestCase):
+    """
+    Test CacheService methods.
+    """
+    @override_settings(CACHES={
+        'default': {
+            'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',
+        }
+    })
+    def test_cache(self):
+        '''
+        Ensure the default cache works as expected.
+        '''
+        cache_service = CacheService(cache)
+        key = 'my_key'
+        value = 'some random value'
+        timeout = 1
+        cache_service.set(key, value, timeout=timeout)
+        assert cache_service.get(key) == value
+        sleep(timeout)
```

Comment:
Would it be helpful to add a bit of a buffer to avoid flakyness or is this guaranteed to always work?

Reply:
Since python3.5, we can rely on `sleep` lasting at least as long as requested ([ref](https://docs.python.org/3/library/time.html#time.sleep)):

> Changed in version 3.5: The function now sleeps at least secs even if the sleep is interrupted by a signal, except if the signal handler raises an exception (see PEP 475 for the rationale).

But I'm happy to add a buffer here if you'd prefer.

Reply:
Sounds good then. 👍🏽